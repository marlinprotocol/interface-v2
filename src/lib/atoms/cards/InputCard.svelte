<script lang="ts">
	import type { InputCardVariant } from '$lib/types/componentTypes';
	import { cn } from '$lib/utils/helpers/commonHelper';
	import { getInputCardVariant } from '$lib/utils/helpers/componentHelper';

	export let styleClass = '';
	export let variant: InputCardVariant | undefined = 'primary';
	export let id = '';

	$: variantClass = getInputCardVariant(variant);
</script>

<div {id} data-testid="input-card" class={cn('w-full rounded-lg', variantClass, styleClass)}>
	<slot />
</div>
