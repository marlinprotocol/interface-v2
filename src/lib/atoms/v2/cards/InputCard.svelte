<script lang="ts">
	import type { InputCardVariant } from '$lib/types/componentTypes';
	import { getInputCardVariant } from '$lib/utils/v2/helpers/componentHelper';

	export let styleClass = '';
	export let variant: InputCardVariant | undefined = 'primary';

	$: variantClass = getInputCardVariant(variant);
</script>

<div data-testid="input-card" class="{styleClass} w-full {variantClass}">
	<slot />
</div>
