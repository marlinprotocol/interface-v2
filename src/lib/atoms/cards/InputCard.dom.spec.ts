import { render } from '@testing-library/svelte';
import { describe, it, expect } from 'vitest';
import InputCard from './InputCard.svelte';
import html from '@playpilot/svelte-htm';

describe('InputCard component', () => {
	it('applies the correct classes for primary variant', () => {
		const { container } = render(InputCard, { variant: 'primary' });
		expect(container).toMatchSnapshot();
	});

	it('applies the correct classes for primary variant', () => {
		const { getByTestId } = render(InputCard, { variant: 'primary' });
		expect(getByTestId('input-card').className).contain('bg-white px-6 py-[30px]');
	});

	it('applies the correct classes for primary-mini variant', () => {
		const { getByTestId } = render(InputCard, { variant: 'primary-mini' });
		expect(getByTestId('input-card').className).contain('bg-base-200 py-2 px-4');
	});

	it('applies the correct classes for warning variant', () => {
		const { getByTestId } = render(InputCard, { variant: 'warning' });
		expect(getByTestId('input-card').className).contain('bg-red-100 px-4 py-1');
	});

	it('applies no classes for none variant', () => {
		const { getByTestId } = render(InputCard, { variant: 'none' });
		expect(getByTestId('input-card').className).contain('');
	});

	it('applies the correct classes for search variant', () => {
		const { getByTestId } = render(InputCard, { variant: 'search' });
		expect(getByTestId('input-card').className).contain('bg-white px-4 py-2.5');
	});

	it('applies default classes when variant is undefined', () => {
		const { getByTestId } = render(InputCard, { variant: undefined });
		expect(getByTestId('input-card').className).contain('w-full bg-white px-6');
	});

	it('should render component inside slot properly', async () => {
		const { findByText } = render(html`
                <${InputCard}>
                 <div>content</div>
                </${InputCard}>
              `);
		expect(await findByText('content')).toBeTruthy();
	});
});
