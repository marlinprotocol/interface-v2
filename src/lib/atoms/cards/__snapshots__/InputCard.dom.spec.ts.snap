// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputCard component > applies the correct classes for primary variant 1`] = `
<div>
  <div
    class="w-full bg-base-100 px-6 py-[30px] rounded-3xl border border-grey-100"
    data-testid="input-card"
  />
</div>
`;
